libtransact/src/contract/archive/mod.rs

```diff
@@ -189,13 +189,30 @@ impl SmartContractArchive {
                 file_path.display()
             ))
         })?;
-        let metadata = metadata.ok_or_else(|| {
+        let metadata: SmartContractMetadata = metadata.ok_or_else(|| {
             Error::new(&format!(
                 "invalid scar file: manifest.yaml not found in {}",
                 file_path.display()
             ))
         })?;
 
+        let version_req = VersionReq::parse(version)?;
+        let metadata_version = Version::parse(&metadata.version).map_err(|err| {
+            Error::new_with_source(
+                &format!(
+                    "invalid metadata version: {}, unable to parse into semver string",
+                    &metadata.version
+                ),
+                err.into(),
+            )
+        })?;
+        if !version_req.matches(&metadata_version) {
+            return Err(Error::new(&format!(
+                "scar file version {} does not match manifest.yaml version {}",
```

Comment:
Can you put the versions in quotes? And s/manifest.yaml/manifest/

Reply:
Addressed